lib/Site.js

```diff
@@ -356,24 +363,39 @@ Site.prototype.generatePages = function () {
   // Render the final rendered page to the output folder.
   const { baseUrl } = this.siteConfig;
   const builtFiles = {};
-  const pages = this.siteConfig.pages || [];
   const processingFiles = [];
+  const pages = this.filePaths || [];
   this.pageModels = pages.map(page => this.createPageData({
     baseUrl,
-    pageSrc: page.src,
-    title: page.title,
+    pageSrc: page,
     pageTemplate: this.pageTemplate,
+    titlePrefix: this.siteConfig.prefix ? this.siteConfig.prefix + TITLE_PREFIX_SEPARATOR : '',
   }));
   this.pageModels.forEach((page) => {
     processingFiles.push(page.generate(builtFiles));
   });
   return new Promise((resolve, reject) => {
     Promise.all(processingFiles)
-      .then(resolve)
-      .catch(reject);
+      .then((frontMatterData) => {
+        this.frontMatterData = frontMatterData;
+        resolve();
+      })
+      .catch((err) => {
+        reject(err);
```

Comment:
can do a 
```js
 .then(resolve)
 .catch(reject);
```
instead of resolving in line 381

If i'm not wrong, line 379 can also be changed to 
```js
.then(frontMatterData => this.frontMatterData = frontMatterData)
```

Reply:
The change can't be made to line 379 as it'll be an error according to ESLint. [no-return-assign](https://eslint.org/docs/rules/no-return-assign)